ta/remoteproc/include/elf_parser.h

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ */
+
+#ifndef ELF_PARSER
+#define ELF_PARSER
+
+#include <elf32.h>
+#include <stdint.h>
+#include <tee_api_types.h>
+
+/**
+ * struct resource_table - firmware resource table header
+ * @ver: version number
+ * @num: number of resource entries
+ * @reserved: reserved (must be zero)
+ * @offset: array of offsets pointing at the various resource entries
+ *
+ * A resource table is essentially a list of system resources required
+ * by the remote processor. It may also include configuration entries.
+ * If needed, the remote processor firmware should contain this table
+ * as a dedicated ".resource_table" ELF section.
+ *
+ * This structure shall be consistent with the Linux kernel structure
+ * definition from include/linux/remoteproc.h.
+ */
+struct resource_table {
+	uint32_t ver;
+	uint32_t num;
+	uint32_t reserved[2];
+	uint32_t offset[];
+} __packed;
```

Comment:
Why `__packed`?

Reply:
This table is defined in a shared memory and used by OPTEE but also the remote processor firmware. 
To avoid alignement issue we pack it. 
Same is done in linux:
https://elixir.bootlin.com/linux/latest/source/include/linux/remoteproc.h#L73

Reply:
Fair enough, I'll skip my usual rant against `__packed` this time. But I'd appreciate a short comment motivating the `__packed` usage to avoid people copying this without understanding why. For instance that it's required by the ELF binary format, interoperability, or something along that.

Reply:
I've added a comment. I hope this is enough to clarify usage.